### PR TITLE
[efsw] Add new port

### LINKED ISSUE #11894: [New Port Request] efsw
Library name: efsw

Library description: File System Watcher

Source repository URL: https://github.com/SpartanJ/efsw

Project homepage (if different from the source repository): 

Anything else that is useful to know when adding (such as optional features the library may have that should be included):

### PR DESCRIPTION
**Pull request by Pospelove**
closes #11894

### CODE REVIEW

File: ports/efsw/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO SpartanJ/efsw
+    REF caacfa665ae6b4da265e7bf28fdf5bc8d343581e
+    SHA512 23458ba287c220a682e39acfd2782c863ea9ca746740579b0310a5d8d9fc1d8a49b4f618a7d18bdab621b6808ee25255f651cc27bcbdd8ebd47003ace021570d
+    HEAD_REF master
+    PATCHES
+    fix_dynamic_build.patch
+)
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
```
Comment (LilyWangL) [inline ports/efsw/portfile.cmake:11]: This is already the default behavior, `vcpkg_configure_cmake` already has:

```
    if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
        list(APPEND _csc_OPTIONS -DBUILD_SHARED_LIBS=ON)
    elseif(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
        list(APPEND _csc_OPTIONS -DBUILD_SHARED_LIBS=OFF)
    else()
        message(FATAL_ERROR
            "Invalid setting for VCPKG_LIBRARY_LINKAGE: \"${VCPKG_LIBRARY_LINKAGE}\". "
            "It must be \"static\" or \"dynamic\"")
    endif()
```

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/efsw/portfile.cmake (hunk 1 above)
Comment (Pospelove) [reply ports/efsw/portfile.cmake]: Fixed

**Comment by LilyWangL** [comment]
Now this port build failed on triplet UWP and MACOS, you can download the error log from the following URL:

- x64-uwp: https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/38877/artifacts?artifactName=x64-uwp%20port%20build%20failure%20logs&api-version=5.1&%24format=zip
- arm-uwp: https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/38877/artifacts?artifactName=x64-uwp%20port%20build%20failure%20logs&api-version=5.1&%24format=zip
- x64-osx: https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/38877/artifacts?artifactName=x64-uwp%20port%20build%20failure%20logs&api-version=5.1&%24format=zip

**Comment by Pospelove** [comment]
I have opened an issue about UWP in source repo (https://github.com/SpartanJ/efsw/issues/108). For now, assume that UWP is not supported.

**Comment by Pospelove** [comment]
It seems there is a typo in the library that causes build failure on MacOS (https://github.com/SpartanJ/efsw/pull/107). Let's wait for the author.

**Comment by Pospelove** [comment]
Ready

**Review by LilyWangL — APPROVED** [review]

**Comment by vicroms** [comment]
Thanks for the PR!

**Pull request merged**